lib/bm-response.js

```diff
@@ -0,0 +1,58 @@
+/* @flow */
+'use strict'
+
+/* ::
+import type {BmRequest, Headers} from './wrapper.js'
+*/
+
+const privateVars = new WeakMap()
+
+class BmResponse {
+  constructor (
+    request /* : BmRequest */
+  ) {
+    privateVars.set(this, {
+      headers: {},
+      payload: undefined,
+      request,
+      statusCode: 200
+    })
+  }
+
+  get headers () /* : Headers */ {
+    return privateVars.get(this).headers
+  }
+
+  get payload () /* : any */ {
+    return privateVars.get(this).payload
+  }
+
+  get statusCode () /* : number */ {
+    return privateVars.get(this).statusCode
+  }
+
+  setHeader (
+    key /* : string */,
+    value /* : string */
+  ) /* : BmResponse */ {
+    key = key.toLowerCase()
+    privateVars.get(this).headers[key] = value
+    return this
```

Comment:
I usually don't put much stock in chainable methods, but I think is makes sense in this case. It certainly makes what is likely to be a fairly common case much nicer for developers. Good thinking. :)